06_puppy/pfannerj/storer_test.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"testing"
+
+	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storerImpl int
+
+const (
+	syncImpl storerImpl = iota // 0
+	mapImpl                    // 1
+)
+
+const firstPuppyID uint32 = 1
+
+var (
+	firstPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Retriever",
+			Colour: "Golden",
+			Value:  9999.99,
+		}
+	}
+	modifiedPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Bitsa",
+			Colour: "Mixed",
+			Value:  1.99,
+		}
+	}
+	anotherPuppy = func() Puppy {
+		return Puppy{
+			Breed:  "Labrador",
+			Colour: "Black",
+			Value:  999.99,
+		}
+	}
+)
+
+type storerSuite struct {
+	suite.Suite
+	store Storer
+	impl  storerImpl
+}
+
+func (s *storerSuite) SetupTest() {
+	var err error
+	switch s.impl {
+	case syncImpl:
+		// create a new syncstore and add the first puppy
+		s.store = NewSyncStore()
+		puppySyncImpl := firstPuppy()
+		_, err = s.store.CreatePuppy(&puppySyncImpl)
+	case mapImpl:
+		// create a new mapstore and add the first puppy
+		s.store = NewMapStore()
+		puppyMapImpl := firstPuppy()
+		_, err = s.store.CreatePuppy(&puppyMapImpl)
+	default:
+		panic("Unrecognised storer implementation")
+	}
+	if err != nil {
+		panic("Failed to setup puppy test")
+	}
+}
+
+func (s *storerSuite) TestCreatePuppy_Successful() {
```

Comment:
Go naming conventions are not to use `_`.
My personal convention is: `TestCreate` and `TestCreateFail` (all of these are methods, you don't have to worry about name clashes within the package.